### DISCUSSION TITLE
My website edge function throws error while my website has no deployments for a long time

### DISCUSSION THREAD

**Discussion by liuxuan30** [vercel/community#7431] [Bug Report] [answered]
### Page to Investigate

https://www.liuxuan30.com/projects/featured-reverse-eng-ios-daemon-journey

### Steps to Reproduce

just open the page, I have a page counter, which code is:
```
export default async function incr(req: NextRequest): Promise<NextResponse> {
	if (req.method !== "POST") {
		return new NextResponse("use POST", { status: 405 });
	}
	if (req.headers.get("Content-Type") !== "application/json") {
		return new NextResponse("must be json", { status: 400 });
	}

	const body = await req.json();
	let slug: string | undefined = undefined;
```

The dashboard gives me logs:
```
JUL 11 14:36:29.19
POST
---
www.liuxuan30.com
/api/incr
SyntaxError: Unexpected end of JSON input at (pages/api/incr.ts:17:14) at (node_modules/next/dist/esm/server/web/adapter.js:103:0)
JUL 11 14:36:29.15
POST
405
www.liuxuan30.com
/api/incr
[POST] /api/incr reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true
JUL 11 14:36:28.87
GET
200
www.liuxuan30.com
/projects/featured-reverse-eng-ios-daemon-journey
JUL 11 14:36:13.33
GET
200
www.liuxuan30.com
/projects
JUL 11 14:36:13.28
POST
---
www.liuxuan30.com
/api/incr
SyntaxError: Unexpected end of JSON input at (pages/api/incr.ts:17:14) at (node_modules/next/dist/esm/server/web/adapter.js:103:0)
JUL 11 14:36:13.27
POST
405
www.liuxuan30.com
/api/incr
[POST] /api/incr reason=EDGE_FUNCTION_INVOCATION_FAILED, status=500, user_error=true
```
it seems the POST API stops working. I haven't touch the code or deploy for a long time. It has been working all the time. Please help.

**Comment by liuxuan30** [comment]
I have tested my code on my local machine with prodctuion credentials, it could update the counter correctly from 115 to 116.  however, the official URL cannot update the counter.

So It seems both the upstash redis API and my code is solid. The issue is on the vercel server side.

**Comment by liuxuan30** [comment]
<img width="1445" alt="image" src="https://github.com/vercel/community/assets/4375169/db4c5002-c0f5-4361-9172-4a1599938709">
proof of working above.

I use production keys to debug this function, all working. No idea why it reports above errors.

**Comment by liuxuan30** [comment]
`const body = await req.json();` the above errors says 'Unexpected end of JSON input at (pages/api/incr.ts:17:14)',

but the value is simply:
```
body
{slug: 'featured-reverse-eng-ios-daemon-journey'}
```

The incr caller is:
```
export const ReportView: React.FC<{ slug: string }> = ({ slug }) => {
	useEffect(() => {
		fetch("/api/incr", {
			method: "POST",
			headers: {
				"Content-Type": "application/json",
			},
			body: JSON.stringify({ slug }),
		});
	}, [slug]);

	return null;
};
```

**Comment by liuxuan30** [comment]
I do see below errors, but I haven't deployed for a long time. Is this related? My errors does not contain 'INTERNAL' but 'EDGE_FUNCTION_INVOCATION_FAILED'
<img width="969" alt="image" src="https://github.com/vercel/community/assets/4375169/cd82e174-f77e-4e18-93f2-c1453492043c">

**Reply by pawlean** [reply]
> I do see below errors, but I haven't deployed for a long time.

It could be related, have you tried deploying again?

**Reply by liuxuan30** [reply]
Hi there,

Not yet;
I read the report, it seems only affecting new deployments during the period, but my current deployment is 1 year ago. So I don't want to deploy again and want to see if anything else.

**Reply by liuxuan30** [reply]
let me try a new deployment first.

**Reply by liuxuan30** [reply, marked as answer]
New deployments fixed the issue. Could you tell me why my site is impacted? 
It has been online for a year and nothing changed. I don't get it why I'm impacted nor the error message could suggest such a fix.

**Reply by pawlean** [reply]
I'll pass this on internally as feedback, and get back to you! Thanks for your patience.

**Comment by Kikobeats** [comment]
Hello, @liuxuan30 the best way for us to reproduce and undestand what's happening is to create a repo reproduction.

Can you create a GitHub repo pushing the minimal code to reproduce the code?

It should to use the same dependencies (Next.js, upstash) with the same version 🙏

**Reply by liuxuan30** [reply]
hi @Kikobeats , sorry for the late reply.

It's hard for me to reproduce as, after you deploy a new one, the old one is offline.

I'd rather hope we move on as this is solved by pushing a new build, which means definitely your server patches impacted the old deployed builds. You just cannot figure out why, and it's not worthy the time to look into.

[a comment by an automated account is not shown]
